dsls/sql/src/main/java/org/apache/beam/dsls/sql/rule/BeamJoinRule.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.dsls.sql.rule;
+
+import org.apache.beam.dsls.sql.rel.BeamJoinRel;
+import org.apache.beam.dsls.sql.rel.BeamLogicalConvention;
+import org.apache.calcite.plan.Convention;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.convert.ConverterRule;
+import org.apache.calcite.rel.core.Join;
+import org.apache.calcite.rel.logical.LogicalJoin;
+
+/**
+ * {@code ConverterRule} to replace {@code Join} with {@code BeamJoinRel}.
+ */
+public class BeamJoinRule extends ConverterRule {
+  public static final BeamJoinRule INSTANCE = new BeamJoinRule();
+  private BeamJoinRule() {
+    super(LogicalJoin.class, Convention.NONE,
```

Comment:
because the first param of `ConverterRule`(`BeamJoinRule`'s parent) is(I quote):

```
Type of relational expression to consider converting
```

For a join, before convert, it's relation algebra tree is something like:

```bash
LogicalProject(order_id=[$0], sum_site_id=[$1], order_id0=[$2], sum_site_id0=[$3])
  LogicalJoin(condition=[=($0, $2)], joinType=[inner])
    LogicalProject(order_id=[$0], sum_site_id=[$2])
      LogicalAggregate(group=[{0, 1}], sum_site_id=[SUM($2)])
        LogicalProject(order_id=[$0], $f1=[TUMBLE($3, 3600000)], site_id=[$1])
          LogicalTableScan(table=[[ORDER_DETAILS]])
    LogicalProject(order_id=[$0], sum_site_id=[$2])
      LogicalAggregate(group=[{0, 1}], sum_site_id=[SUM($2)])
        LogicalProject(order_id=[$0], $f1=[TUMBLE($3, 3600000)], site_id=[$1])
          LogicalTableScan(table=[[ORDER_DETAILS]])
```

Note the `LogicalJoin` in the tree, it is the class we are trying to convert to `BeamJoinRel`.